src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocleadingasteriskalign/InputJavadocLeadingAsteriskAlignIncorrect.java

```diff
@@ -0,0 +1,81 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="JavadocLeadingAsteriskAlign" />
+  </module>
+</module>
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocleadingasteriskalign;
+
+/**
+* This is the class level javadoc. // violation
+ */
+public class InputJavadocLeadingAsteriskAlignIncorrect {
+  /**
+    * Javadoc for instance variable // violation
+   */
+  private int age;
+
+  /**
+  *  // violation
+   */
+  private String name;
+
+  /**
+   * Javadoc for foo.
+    */ // violation
+  public void foo() {}
+
+  /**
+  */ // violation
+  public void foo2() {}
+
+  /**
+      * // violation
+   */
+  public void foo3() {}
+
+  /**
+*   // violation
+   */
+  public void foo4() {}
+
+  /**
+   * Default Constructor.
+      */ // violation
+  public InputJavadocLeadingAsteriskAlignIncorrect() {}
+
+  /**
+   * Parameterized Constructor.
+*/ // violation
```

Comment:
It should be added to the documentation that the ending `*/` is considered part of a leading, even if there is stuff before it.

Reply:
> even if there is stuff before it.

No this is not true. The checking for alignment of the endling line of javadoc depends on various  situations. I've explained them here: https://github.com/checkstyle/checkstyle/pull/14643#discussion_r1583626034

Reply:
ok, this should be covered by the other for some written form of it.

Reply:
done. updated the check description mentioning this:

https://github.com/checkstyle/checkstyle/blob/0c885e85b4ffc6c87ba4edb32d72f52165b70606/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocLeadingAsteriskAlignCheck.java#L41-L42